api/src/main/java/org/pentaho/platform/api/email/IEmailService.java

```diff
@@ -39,11 +46,37 @@ public interface IEmailService {
 
   /**
    * Generates a test email via the specficied email configuration
-   * 
+   *
    * @param emailConfig
    *          the email configuration to use for sending the testing email
    */
   public String sendEmailTest( final IEmailConfiguration emailConfig );
 
   public boolean isValid();
+
+  /**
+   * Method used to obtain a new Access Token during sending mail when Authentication is OAuth based
+   *
+   * @param emailConfig
+   *          the email configuration to use for sending the email
+   *
+   */
+  public IEmailAuthenticationResponse getOAuthToken( final IEmailConfiguration emailConfig ) throws IOException, HttpException;
+
+
+  /**
+   * Sends a mail using SMTP or Graph API depending on Params
+   *
+   * @param emailConfig
+   *          the email configuration to use for sending the email
+   * @param token
+   *          the token object containing the access token required for OAuth
+   * @param session
+   *          the java mail session required for SMTP connections
+   * @param msg
+   *          the message object in form of MimeMessage. Sent as object for SMTP and as Base64 string in Graph API calls
+   *
+   */
+  public void sendEmail( final IEmailConfiguration emailConfig, IEmailAuthenticationResponse token, Session session, MimeMessage msg ) throws MessagingException, IOException, HttpException;
```

Comment:
To make your interface definition cleaner, it's common to create  a "wrapper" exception that will handle all the checked exceptions. This eliminates having to grow your throws part of the function for every kind of checked exception. Unless, the caller/client of this interface does something different based on the type exception thrown. But from the current code base, it has a generic `catch (Exception e )`

example: https://github.com/NJtwentyone/pentaho-platform/blob/master/extensions/src/main/java/org/pentaho/platform/config/AppConfigException.java

- not sure about should you just extends directly `Exception` instead of `CheckedException`

```suggestion
  public void sendEmail( final IEmailConfiguration emailConfig, IEmailAuthenticationResponse token, Session session, MimeMessage msg ) throws EmailServiceException;
```

then in the code EmailService
```
public void sendEmail(....) {
   ... ... ...
   
     } catch ( MessagingException  me ) {
       throw EmailServiceException("some message about message exception" , me);
     }
  }
```